android-kotlin/app/src/main/java/com/example/msaldemo/MainActivity.kt

```diff
@@ -0,0 +1,184 @@
+package com.example.msaldemo
+
+// Required dependencies for Android
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+
+// Required dependencies for MSAL
+import com.microsoft.identity.client.*
+import com.microsoft.identity.client.exception.MsalException
+
+// Required dependencies for Volley - Used for making HTTP requests to Graph
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.Volley
+
+// Required dependency for working with JSON objects
+import org.json.JSONObject
+
+class MainActivity : AppCompatActivity() {
+    private var msalApplication: ISingleAccountPublicClientApplication? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.content_main)
+
+        // Creates a Public Application using MSAL configuration for single account access
+        PublicClientApplication.createSingleAccountPublicClientApplication(
+            this,
+            R.raw.msal_auth_config,
+            object : IPublicClientApplication.ISingleAccountApplicationCreatedListener {
+                override fun onCreated(application: ISingleAccountPublicClientApplication) {
+                    msalApplication = application
+                    loadAccount()
+                }
+
+                override fun onError(exception: MsalException?) {
+                    print(exception.toString())
+                }
+            })
+    }
+
+    private fun initializeUI() {
+        val btnSignIn: Button = findViewById(R.id.btn_signIn)
+        val btnRemoveAccount: Button = findViewById(R.id.btn_signOut)
+        val btnCallGraph: Button = findViewById(R.id.btn_callGraph)
+
+        // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+        val authority = ""
```

Comment:
will give it a shot

Reply:
Love this suggestion btw.  Got it to work.  Not sure how optimal it is.

Reply:
Updated.

Reply:
Heh, I see what you mean. To me it's so weird to think that the data isn't accessible in the `msalApplication`. It was handed that information, really weird it just hides it.

Well since it looks like you're saying it's not in the msal client anywhere, and this is the code to read it from the resource, probably better to just leave it like you had it.

Reply:
@ckittel fine, I guess we can do it the easy way 😆 ..doh! 🤦 

Updated

Reply:
Looks so much nicer now.  Thanks!